src/pages/settings/Payments/AddDebitCardPage.js

```diff
@@ -0,0 +1,228 @@
+            return false;
+        }
+
+        if (!/^[0-9]{3,4}$/.test(this.state.securityCode)) {
+            Growl.error(this.props.translate('addDebitCardPage.error.securityCode'));
+            return false;
+        }
+
+        if (!isValidAddress(this.state.billingAddress)) {
+            Growl.error(this.props.translate('addDebitCardPage.error.address'));
+            return false;
+        }
+
+        if (this.state.city === '') {
+            Growl.error(this.props.translate('addDebitCardPage.error.addressCity'));
+            return false;
+        }
+
+        if (this.state.selectedState === '') {
+            Growl.error(this.props.translate('addDebitCardPage.error.addressState'));
+            return false;
+        }
+
+        if (!isValidZipCode(this.state.zipCode)) {
+            Growl.error(this.props.translate('addDebitCardPage.error.zipCode'));
+            return false;
+        }
+
+        if (!this.state.acceptedTerms) {
+            Growl.error(this.props.translate('addDebitCardPage.error.acceptedTerms'));
+            return false;
+        }
+
+        return true;
+    }
+
+    submit() {
+        if (!this.validate()) {
+            return;
+        }
+        addBillingCard(this.state);
+    }
+
+    toggleTermsOfService() {
+        this.setState(prevState => ({acceptedTerms: !prevState.acceptedTerms}));
+    }
+
+    render() {
+        return (
+            <ScreenWrapper>
+                <KeyboardAvoidingView>
+                    <HeaderWithCloseButton
+                        title={this.props.translate('addDebitCardPage.addADebitCard')}
+                        onCloseButtonPress={() => Navigation.dismissModal(true)}
+                    />
+                    <ScrollView style={styles.flex1} contentContainerStyle={styles.p5}>
+                        <TextInputWithLabel
+                            label={this.props.translate('addDebitCardPage.nameOnCard')}
+                            placeholder={this.props.translate('addDebitCardPage.nameOnCard')}
+                            containerStyles={[styles.flex1, styles.mb2]}
+                            onChangeText={nameOnCard => this.setState({nameOnCard})}
+                            value={this.state.nameOnCard}
+                        />
+                        <TextInputWithLabel
+                            label={this.props.translate('addDebitCardPage.debitCardNumber')}
+                            placeholder={this.props.translate('addDebitCardPage.debitCardNumber')}
+                            keyboardType="number-pad"
+                            containerStyles={[styles.flex1, styles.mb2]}
+                            onChangeText={cardNumber => this.setState({cardNumber})}
+                            value={this.state.cardNumber}
+                        />
+                        <View style={[styles.flexRow, styles.mb2]}>
+                            <TextInputWithLabel
```

Comment:
Can we make it so that when you're typing we automatically put in a `/` after the first 2 digits? Not sure how hard that would be to implement tbh, but I think it's a big quality of life improvement

Reply:
Agreed! Good idea